### DISCUSSION TITLE
Hooks that do something when certain Homebrew event happens?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#1845] [Everyday usage]
## Why?

I always check the hash of downloaded casks on virustotal.com (VT) before opening them. 

This is how I usually do it:
- If the sha256sum is provided in the cask file, I'd search that hash on VT.
- If the checksum is not provided (`sha256 :no_check`), I'd search the hash of the file at this path: 
`"$(brew --cache --cask <cask>)"`.

If brew cask supports `hooks`, this can be easily automated, e.g. when a cask is successfully installed / upgraded, the hook runs a script that does the VT check for me. (I already wrote the script, and just need it to run on these events.)

---

Personally I think hook support is a good idea, although it doesn't exactly fit into Homebrew's core functionalities.

What do you think of hooks in Homebrew?

**Comment by SMillerDev** [comment]
I would happily review a pull request for it.

**Comment by wfiveash** [comment]
I would also like hook support for homebrew actions for similar reasons.
